.github/workflows/update-api-files.yml

```diff
@@ -37,10 +37,17 @@ jobs:
           git config user.name "github-actions[bot]"
       # If run from a PR, checkout PR branch and merge in latest changes from main branch
       - name: Checkout PR branch and merge main
+        id: checkout-pr-branch
         if: ${{ github.event_name != 'workflow_dispatch' }}
-        run: |
```

Comment:
There is another approach we can do as well. 
Instead of using builtin github commands (Which I prefer here), we can instead sanitize the branch directly.

```bash
- name: Checkout PR branch and merge main
        if: ${{ github.event_name != 'workflow_dispatch' }}
        run: |
            PR_BRANCH="${{ github.event.pull_request.head.ref }}"
            SANITIZED_PR_BRANCH=$(echo "$PR_BRANCH" | sed 's/[^a-zA-Z0-9_\-\/]//g')
            git fetch origin
            git checkout "$SANITIZED_PR_BRANCH"
            git merge origin/main
 ```